widget/embedded/src/pages/Home.tsx

```diff
@@ -95,204 +60,146 @@ export function Home() {
     warning: quoteWarning,
     needsToWarnEthOnPath,
   });
-
-  const fromTokenBalance = fromToken ? getBalanceFor(fromToken) : null;
-
-  const fromTokenFormattedBalance =
-    formatBalance(fromTokenBalance)?.amount ?? '0';
-
-  const tokenBalanceReal =
-    !!fromBlockchain && !!fromToken && fromTokenBalance?.amount
-      ? numberToString(fromTokenBalance?.amount, fromTokenBalance?.decimals)
-      : '0';
-
-  const fetchingQuote =
-    fetchMetaStatus === 'success' &&
-    !!inputAmount &&
-    !!fromToken &&
-    !!toToken &&
-    loading;
+  const isExpandable = config?.variant === 'expandable';
+  const hasInputs =
+    !!inputAmount && !!fromToken && !!toToken && inputAmount !== '0';
+  const fetchingQuote = hasInputs && fetchMetaStatus === 'success' && loading;
+  const showMessages =
+    (!isExpandable || (isExpandable && quotes?.results.length)) &&
+    (quoteWarning || quoteError);
 
   useEffect(() => {
     resetQuoteWallets();
     updateQuotePartialState('refetchQuote', true);
   }, []);
 
-  const percentageChange =
-    !inputUsdValue || !outputUsdValue || !outputUsdValue.gt(0)
-      ? null
-      : getPriceImpact(inputUsdValue.toString(), outputUsdValue.toString());
+  useEffect(() => {
+    setIsVisibleExpanded(hasInputs);
+  }, [hasInputs]);
+
+  const onClickRefresh =
+    (!!selectedQuote || quoteError) && !showQuoteWarningModal
+      ? fetchQuote
+      : undefined;
+
+  const onHandleNavigation = (route: string) => {
+    if (isExpandable) {
```

Comment:
Should we also check isVisibleExpanded? Navigating still takes time when there is no input.

Reply:
yes, you're write